lib/spack/spack/spec.py

```diff
@@ -669,8 +669,8 @@ def __str__(self):
         sorted_keys = [k for k in sorted(self.keys()) if self[k] != []]
         cond_symbol = ' ' if len(sorted_keys) > 0 else ''
         return cond_symbol + ' '.join(
-            str(key) + '=\"' + ' '.join(
```

Comment:
Because we no longer use escape quotes on the command line, and the string representation only had them to match the command line syntax. This was really a change that was missed in a previous PR of mine that I noticed while working on this one.